static/js/components/UserManagement.jsx

```diff
@@ -0,0 +1,198 @@
+      dispatch(usersActions.fetchUsers());
+    }
+  };
+
+  const handleClickAddUsers = async () => {
+    let rows = PapaParse.parse(csvData.trim(), {
+      delimiter: ",",
+      skipEmptyLines: "greedy",
+    }).data;
+    rows = rows.map((row) => [
+      row[0].trim(),
+      PapaParse.parse(row[1].trim(), { delimiter: " " }).data[0],
+      PapaParse.parse(row[2].trim(), { delimiter: " " }).data[0],
+      PapaParse.parse(row[3].trim(), { delimiter: " " }).data[0],
+    ]);
+    const promises = rows.map((row) =>
+      dispatch(
+        inviteUsersActions.inviteUser({
+          userEmail: row[0],
+          streamIDs: row[1],
+          groupIDs: row[2],
+          groupAdmin: row[3],
+        })
+      )
+    );
+    const results = await Promise.all(promises);
+    if (results.every((result) => result.status === "success")) {
+      dispatch(showNotification("User(s) successfully invited."));
+      setCsvData("");
+    }
+  };
+
+  return (
+    <>
+      <Typography variant="h5">Manage users</Typography>
+      <Paper elevation={1}>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>Name</TableCell>
+              <TableCell>Username</TableCell>
+              <TableCell>Email</TableCell>
+              <TableCell>Groups</TableCell>
+              <TableCell>Streams</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {allUsers.map((user) => (
+              <TableRow key={user.id}>
+                <TableCell>
+                  {`${user.first_name ? user.first_name : ""} ${
+                    user.last_name ? user.last_name : ""
+                  }`}
+                </TableCell>
+                <TableCell>{user.username}</TableCell>
+                <TableCell>{user.contact_email}</TableCell>
+                <TableCell>
+                  {user.groups
+                    .filter((group) => !group.single_user_group)
+                    .map((group) => (
+                      <Chip
+                        label={group.name}
+                        onDelete={() => {
+                          handleClickRemoveUserFromGroup(
+                            user.username,
+                            group.id
+                          );
+                        }}
+                        key={group.id}
+                        id={`deleteGroupUserButton_${user.id}_${group.id}`}
+                      />
+                    ))}
+                </TableCell>
+                <TableCell>
+                  {user.streams.map((stream) => (
+                    <Chip
+                      label={stream.name}
+                      onDelete={() => {
+                        handleClickRemoveUserStreamAccess(user.id, stream.id);
+                      }}
+                      key={stream.id}
+                      id={`deleteStreamUserButton_${user.id}_${stream.id}`}
+                    />
+                  ))}
+                </TableCell>
```

Comment:
Let's leave it for another PR, but would also be nice to be able to add stream access here as well.

Reply:
Yes, and also perhaps add existing users to groups

Reply:
Will capture this in an issue/story